core/src/main/java/org/apache/any23/validator/rule/MissingItemscopeAttributeValueRule.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.any23.validator.rule;
+
+import org.apache.any23.validator.DOMDocument;
+import org.apache.any23.validator.Fix;
+import org.apache.any23.validator.Rule;
+import org.apache.any23.validator.RuleContext;
+
+/**
+ * This fixes missing attribute values for the 'itemscope' attribute, 
+ * which was be associated with <div> nodes.
+ * Typically when such a snippet of XHTML is fed through the 
+ * {@link org.apache.any23.extractor.rdfa.RDFa11Extractor}, and
+ * subsequently to Sesame's {@link org.semarglproject.sesame.rdf.rdfa.SesameRDFaParser},
+ * it will result in the following behavior. 
+ * <pre>
+ * {@code
+ * [Fatal Error] :23:15: Attribute name "itemscope" associated with an element type "div" must be followed by the ' = ' character.
+ * }
+ * </pre>
+ * This Fix is an effort to mitigate against that happening. 
+ *
+ */
+public class MissingItemscopeAttributeValueRule implements Fix {
```

Comment:
Ack

On Monday, March 30, 2015, Peter Ansell <notifications@github.com> wrote:

> In
> core/src/main/java/org/apache/any23/validator/rule/MissingItemscopeAttributeValueRule.java
> <https://github.com/apache/any23/pull/17#discussion_r27442717>:
>
> > +/**
> > + * This fixes missing attribute values for the 'itemscope' attribute,
> > + * which was be associated with <div> nodes.
> > + * Typically when such a snippet of XHTML is fed through the
> > + * {@link org.apache.any23.extractor.rdfa.RDFa11Extractor}, and
> > + * subsequently to Sesame's {@link org.semarglproject.sesame.rdf.rdfa.SesameRDFaParser},
> > + * it will result in the following behavior.
> > + * <pre>
> > + * {@code
> > + * [Fatal Error] :23:15: Attribute name "itemscope" associated with an element type "div" must be followed by the ' = ' character.
> > + * }
> > + * </pre>
> > + * This Fix is an effort to mitigate against that happening.
> > + *
> > + */
> > +public class MissingItemscopeAttributeValueRule implements Fix {
>
> It may be done using a classpath scan. I will look into it further.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/any23/pull/17/files#r27442717>.
>


-- 
*Lewis*